client/src/components/InsightsTab.tsx

```
@@ -0,0 +1,122 @@
+import { useState } from 'react';
+import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
+import { Brain } from 'lucide-react';
+import { Button } from './ui/button';
+import { getFinancialInsights } from '@/lib/ai-service';
+import { PropertyType } from '@/types/PropertyTypes';
+import { saveInsights } from '@/lib/ai-service';
+import type { AIInsights } from '@/lib/ai-service';
+
+
+interface AIInsightsProps {
+  properties: PropertyType[];
+  insights: AIInsights | null;
+  isLoading: boolean;
+  onInsightsUpdated: () => Promise<void>;
+}
+
+export function AIInsightsTab({ 
+  properties, 
+  insights, 
+  isLoading, 
+  onInsightsUpdated
+ }: AIInsightsProps) {
+  const [loading, setLoading] = useState(false);
+
+
+
+  async function generateAndSaveInsights() {
+    try {
+      setLoading(true);
+      const result = await getFinancialInsights(properties);
+      
+      if (result && result.insights) {
+        const { overview, timelineToPurchase, marketTrends, peerStrategies } = result.insights;
+
+        if (
+          typeof overview === 'string' &&
+          typeof timelineToPurchase === 'string' &&
+          typeof marketTrends === 'string' &&
+          typeof peerStrategies === 'string'
+        ) {
+          try {
+            await saveInsights(result.insights);
+            await onInsightsUpdated();
+          } catch (saveError) {
+            console.error('Error saving insights:', saveError);
+          }
+        }
```

I expected to see an `else` with error handling. It looks like if `getFinancialInsights` doesn't return a complete response then your code silently does nothing. That is very disconcerting to a user. At minimum, you should display an error message somehow, even if it's just in an alert.